eng/automation/utils.py

```diff
@@ -352,17 +352,13 @@ def set_or_increase_version(
 
     # 1.0.0-beta.1 -> [1,0,0,"-beta.1"]
     current_versions = list(re.findall(version_pattern, current_version)[0])
-    # 1.0.0 -> [(1,0,0,"")]
-    # 1.0.0-beta.1 -> [(1,0,0,"-beta.1")]
-    stable_versions = re.findall(version_pattern, stable_version)
-    # no stable version
-    if len(stable_versions) < 1 or stable_versions[0][-1] != "":
-        if not preview:
-            current_versions[-1] = ""
+    # if not preview, and version is not specified, set current version to be stable version
+    if not preview and not version:
+        current_versions[-1] = ""
         current_version = version_format.format(*current_versions)
         if not stable_version:
             stable_version = current_version
-        logging.info('[VERSION][Not Found] cannot find stable version, current version "{0}"'.format(current_version))
+        logging.info('[VERSION] stable release, auto-set current version to "{0}"'.format(current_version))
```

Comment:
`preview` property is never used before. 
Here, adapt it's meaning to sdkReleaseType.